tests/bsim/bluetooth/mesh/src/test_priv_beacon.c

```diff
@@ -0,0 +1,564 @@
+
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_PHASE_2);
+
+	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+	memcpy(last_random, curr_random, 13);
+
+	wait_for_next_beacon(BEACON_INTERVAL);
+
+	status = bt_mesh_subnet_kr_phase_get(BT_MESH_KEY_PRIMARY, &phase);
+	ASSERT_TRUE(status == STATUS_SUCCESS);
+	ASSERT_TRUE(phase == BT_MESH_KR_NORMAL);
+}
+
+static void test_rx_on_kr(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 1;
+
+	err = bt_mesh_priv_beacon_cli_set(0, RX_ADDR + 1, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	commot_test_rx_kr();
+
+	PASS();
+}
+
+static void test_rx_on_kr_long_int(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 3;
+
+	err = bt_mesh_priv_beacon_cli_set(0, RX_ADDR + 1, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	commot_test_rx_kr();
+
+	PASS();
+}
+#endif
+
+#if defined(CONFIG_BT_MESH_GATT_PROXY)
+static void test_tx_proxy(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &tx_comp);
+	prov_and_conf_tx(bt_mesh_test_own_addr_get(RX_ADDR));
+
+	/* shift beaconing time line to avoid boundary cases. */
+	k_sleep(K_SECONDS(1));
+
+	PASS();
+}
+
+static void test_rx_proxy(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err, i;
+
+	common_rx_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 1;
+
+	err = bt_mesh_priv_beacon_cli_set(0, RX_ADDR + 1, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	/* Because expected BEACON_INTERVAL is 10s we expect Random Value to be changed on
+	 * every beacon
+	 */
+	for (i = 0; i < 6; i++) {
+		wait_for_next_beacon(BEACON_INTERVAL);
+
+		ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+		memcpy(last_random, curr_random, 13);
+	}
+
+	val.rand_interval = 0;
+	err = bt_mesh_priv_beacon_cli_set(0, RX_ADDR + 1, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	/* Because expected BEACON_INTERVAL is 10s, this should behave the same as for
+	 * Random Interval = 1
+	 */
+	for (i = 0; i < 6; i++) {
+		wait_for_next_beacon(BEACON_INTERVAL);
+
+		ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+		memcpy(last_random, curr_random, 13);
+	}
+
+	val.rand_interval = 3;
+	err = bt_mesh_priv_beacon_cli_set(0, RX_ADDR + 1, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	beaconing_time = 0;
+
+	for (i = 0; i < 6; i++) {
+		wait_for_next_beacon(BEACON_INTERVAL);
+
+		if (beaconing_time < val.rand_interval * 10 * MSEC_PER_SEC) {
+			LOG_DBG("Random is same, beaconing time %d", beaconing_time);
+			ASSERT_TRUE(memcmp(last_random, curr_random, 13) == 0);
+		} else {
```

Comment:
Can we avoid this if-else? It is better to avoid if-elses in tests to be sure we test what we want. Since the interval is 30 seconds, the first 3 beacons should have the same random, the consecutive 3 beacons should have a new random. Something like this should be better:
```
for (int i = 0; i < 2; i++) {
	wait_for_next_beacon(BEACON_INTERVAL);

	// Beacon random should change here
	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
	memcpy(last_random, curr_random, 13);

	for (int j = 0; j < 2; j++) {
		wait_for_next_beacon(BEACON_INTERVAL);

		// Beacon random should be the same here
		ASSERT_TRUE(memcmp(last_random, curr_random, 13) == 0);
		memcpy(last_random, curr_random, 13);
	}
}

Reply:
Makes sense, it works but needs to be like this:
```c
	for (i = 0; i < 2; i++) {
		for (int j = 0; j < 2; j++) {
			wait_for_next_priv_beacon(BEACON_INTERVAL);

			// Beacon random should be the same here
			ASSERT_TRUE(memcmp(last_random, curr_random, 13) == 0);
		}

		wait_for_next_priv_beacon(BEACON_INTERVAL);

		// Beacon random should change here
		ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
		memcpy(last_random, curr_random, 13);

	}
```